search/collector/topn.go

```diff
@@ -154,6 +154,23 @@ func (hc *TopNCollector) Collect(ctx context.Context, searcher search.Searcher,
 		hc.sort.UpdateVisitor(field, term)
 	}
 
+	var dmHandlerMaker search.MakeDocumentMatchHandler
+	var dmHandler search.DocumentMatchHandler
+	var loadID, ok bool
+	if cv := ctx.Value(search.MakeDocumentMatchHandlerKey); cv == nil {
+		// initialize the dmHandler with the default TopN handler
+		dmHandler =  NewTopNDocumentMatchHandler(searchContext, hc, nil)
+	} else if dmHandlerMaker, ok = cv.(search.MakeDocumentMatchHandler); !ok {
+		// on issues with the given doc match handler maker, 
+		// initialize the dmHandler with the default TopN handler
+		dmHandler =  NewTopNDocumentMatchHandler(searchContext, hc, nil)
+	} else {
+		// use the application given builder for making the custom document match
+		// handler and perform callbacks/invocations on the newly made handler.
+		dmHandler = dmHandlerMaker(searchContext)
+		loadID = true
```

Comment:
the loadID here looks a little like an odd, special case?  Instead, what if the dmHandlerMaker() returned an additional bool loadID return value, so that the callback can decide whether it wants ID's loaded or not?

Reply:
hm, thinking was that IndexInternalID doesn't make sense for applications. We can make it optional too..

Reply:
should the second return value for dmHandlerMaker () should be just the loadID bool or a n options struct?